libcontainer/notify_linux_v2.go

```diff
@@ -0,0 +1,92 @@
+// +build linux
+
+package libcontainer
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+)
+
+func checkOOM(path string) (bool, error) {
+	file, err := os.Open(path)
+	defer file.Close()
+	if err != nil {
+		return false, err
+	}
+	buff := bufio.NewReader(file)
+	for {
+		line, err := buff.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+		arr := strings.Fields(line)
```

Comment:
So, you are only analyzing the last line (which on my system starts with `oom_kill`).

What you should do (I think) is
1 read/parse the file
2 get the counters for `oom` and `oom_kill` lines
3 compare these with the previous readings (or `0` if there are none)
4 save the counters (to be used later on step 3)
5 if counters are increased, return `true`